modules/core/src/main/java/org/apache/ignite/internal/processors/metric/list/walker/CacheGroupViewWalker.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.metric.list.walker;
+
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.cache.CacheRebalanceMode;
+import org.apache.ignite.cache.PartitionLossPolicy;
+import org.apache.ignite.spi.metric.list.MonitoringRowAttributeWalker;
+import org.apache.ignite.spi.metric.list.view.CacheGroupView;
+
+/**
+ * Generated by {@code org.apache.ignite.codegen.MonitoringRowAttributeWalkerGenerator}.
+ * {@link CacheGroupView} attributes walker.
+ * 
+ * @see CacheGroupView
+ */
+public class CacheGroupViewWalker implements MonitoringRowAttributeWalker<CacheGroupView> {
+    /** {@inheritDoc} */
+    @Override public void visitAll(AttributeVisitor v) {
+        v.accept(0, "cacheGroupName", String.class);
```

Comment:
It may irritate in future to support such walkers because of matching id -> name in several methods.
Maybe it makes sense describing fields names in private enum, possibly it makes code cleaner and #count() will return #values().length

```    
    private enum CacheFields {
        CACHE_GROUP_NAME(0, "cacheGroupName"),
        CACHE_COUNT(1, "cacheCount"),
        // etc.;

        private final int idx;
        private final String name;

        CacheFields(int idx, String name) {
            this.idx = idx;
            this.name = name;
        }
    }
 ```

Reply:
Actually, this code was generated by `MonitoringRowAttributeWalkerGenerator`.
You can find its source code in this PR.
So, when you change some view class, you simply run it and got up to date walker implementation.

What do you think?

Reply:
Thanks for the explanation, no problem then.